core/dbt/parser/read_files.py

```diff
@@ -12,13 +12,26 @@
 # This loads the files contents and creates the SourceFile object
 def load_source_file(
         path: FilePath, parse_file_type: ParseFileType,
-        project_name: str) -> Optional[AnySourceFile]:
-    file_contents = load_file_contents(path.absolute_path, strip=False)
-    checksum = FileHash.from_contents(file_contents)
+        project_name: str, saved_files,) -> Optional[AnySourceFile]:
+
     sf_cls = SchemaSourceFile if parse_file_type == ParseFileType.Schema else SourceFile
-    source_file = sf_cls(path=path, checksum=checksum,
+    source_file = sf_cls(path=path, checksum=FileHash.empty(),
                          parse_file_type=parse_file_type, project_name=project_name)
-    source_file.contents = file_contents.strip()
+
+    skip_loading_schema_file = False
+    if (saved_files and source_file.file_id in saved_files and
+            parse_file_type == ParseFileType.Schema):
```

Comment:
nit: suggesting a better short circuit
```suggestion
    if (parse_file_type == ParseFileType.Schema and
    		saved_files and source_file.file_id in saved_files):
```